src/core/Akka/Actor/SupervisorStrategy.cs

```diff
@@ -193,10 +190,9 @@ public static SupervisorStrategy DefaultStrategy
     /// </summary>
     public class OneForOneStrategy : SupervisorStrategy
     {
-        private readonly int _maxNumberOfRetries;
-        private readonly int _withinTimeRangeMilliseconds;
-        private readonly Func<Exception, Directive> _decider;
-        private readonly bool _loggingEnabled;
+        public int MaxNumberOfRetries { get; private set; }
+        public int WithinTimeRangeMilliseconds { get; private set; }
+        public IDecider Decider { get; private set; }
 
```

Comment:
I this is the wrong way to go, this makes a super strong coupling to Json.NET since that is pretty much the only standard serializer that supports this kind of ctor deserialization, while most do support property based deserialization.
This would also mean that every primitive would have to expose a ctor that matches all of its properties. Maybe it supports private/protected ctors? otherwise we would expose an API that is for the serializer only and not intended for end users.
We would also have to verify how this plays with surrogates.

If we do this, it should be applied to all the primitives.
I think its a very high price for the sake of purism.